apps/project/serializers.py

```diff
@@ -389,3 +314,59 @@ def update(self, instance: Organization, validated_data: dict[typing.Any, typing
         organization = super().update(instance, validated_data)
         transaction.on_commit(lambda: FirebaseOrganizationPush(organization.pk).push())
         return organization
+
+
+class ProjectStatusUpdateSerializer(UserResourceSerializer[Project]):
+    class Meta:  # type: ignore[reportIncompatibleVariableOverride]
+        model = Project
+        fields = ("status",)
+
+    def validate_status(self, new_status: Project.Status | int) -> Project.Status:
+        assert self.instance is not None, "Project does not exist."
+
+        if not isinstance(new_status, Project.Status):
+            new_status = Project.Status(new_status)
+
+        if (
+            self.instance.status_enum != new_status
+            and (self.instance.status_enum, new_status) not in VALID_PROJECT_STATUS_TRANSITIONS
+        ):
+            raise serializers.ValidationError(
+                gettext("Project status cannot be changed from %s to %s")
+                % (
+                    self.instance.status_enum.label,
+                    new_status.label,
+                ),
+            )
+
+        if new_status == Project.Status.PUBLISHED and not self.instance.tutorial:
+            raise serializers.ValidationError(
+                {"tutorial": gettext("Tutorial is required before publishing a project.")},
+            )
+        if new_status == Project.Status.MARKED_AS_READY and self.instance.project_type_specifics is None:
```

Comment:
We might also need to check if the json is empty??

```suggestion
        if new_status == Project.Status.MARKED_AS_READY and self.instance.project_type_specifics is None or self.instance.project_type_specifics == {}:
```

<img width="363" height="228" alt="Image" src="https://github.com/user-attachments/assets/5f8bfad8-fc24-44e0-b886-62234e4368ee" />

Reply:
`if new_status == Project.Status.MARKED_AS_READY and not self.instance.project_type_specifics`
Adding `not` supports both cases.